draft-ietf-lpwan-schc-over-lorawan.md

```diff
@@ -394,8 +396,44 @@ The mechanism for sharing those RuleID values is outside the scope of this docum
 
 ## IID computation
 
-It is RECOMMENDED to create Interface IDentifier following
-{{I-D.ietf-lpwan-ipv6-static-context-hc}}, [rfc7217] and [rfc8064]
+In order to mitigate risks described in [rfc8064] and [rfc8065] IID MUST be
+created regarding the following algorithm:
+
+1. key = LoRaWAN AppSKey
+2. string = devEui in HEX representation, padded to 128 bits by 0 on the left
```

Comment:
Change "on the left" to "padded to 128 bits by adding 0 as most significant bits"

Reply:
Hello,

Since IID is only computed on the SCHC gateway (not on the end-device), I don't see the need to rely on (1) algorithms available on the end-device and (2) key material available on the end-device.

I understand the concept of obfuscating the devEUI for privacy [RFC8065]. I would recommend to use a standard cryptographic hash function (say SHA256), and not relying on the security context of the end-device, as it might not be available on the SCHC gateway. If there is a need to refresh the IID after every join, better use the devAddr. Better yet, use a random salt in the hash function to refresh regularly the IID, as LoRaWAN session can be (very) long.

      Specifications should make sure that an IPv6 address can change
      over long periods of time.  For example, the interface identifier
      might change each time a device connects to the network (if
      connections are short) or might change each day (if connections
      can be long).  This is necessary to mitigate correlation over
      time.
     [RFC8065 Section 4]

Reply:
Hello, why do you assume that "IID is only computed on the SCHC gateway (not on the end-device)"? You could potentially have a constant IID at the end-device (that would be elided on the LoRaWAN link) and a privacy-respecting IID on the Internet, but this would mean that the SCHC C/D on the Infrastructure Side would do NAT. Is this what you have in mind ?

Reply:
> Since IID is only computed on the SCHC gateway (not on the end-device),

It is also computed in the device to reconstruct incoming IPv6 pakets (from network to device)

> and not relying on the security context of the end-device, as it might not be available on the SCHC gateway. If there is a need to refresh the IID after every join, better use the devAddr. 

That is why I used the AppSKey instead of the devAddr:
1. devAddr is usually forwarded to application server (SCHC gateway here) but not always.
2. AppSKey is know by the AS (SCHC gateway) and not the NS => Avoid correlation at network level
3. devAddr might not be unique in a network